lib/rubocop/cop/style/trailing_underscore_variable.rb

```diff
@@ -89,35 +91,51 @@ def allow_named_underscore_variables
             cop_config['AllowNamedUnderscoreVariables']
         end
 
-        def unneeded_range(node)
-          left, right = *node
-          variables = *left
+        def unneeded_ranges(node)
+          node.masgn_type? ? (mlhs_node, _ = *node) : mlhs_node = node
+          variables = *mlhs_node
+
+          children_offenses(variables) << main_node_offense(node)
+        end
+
+        def main_node_offense(node)
+          node.masgn_type? ? (mlhs_node, right = *node) : mlhs_node = node
+
+          variables = *mlhs_node
           first_offense = find_first_offense(variables)
 
           return unless first_offense
 
           if unused_variables_only?(first_offense, variables)
-            return left_side_range(left, right)
+            return unused_ranges(node.type, mlhs_node, right)
           end
 
-          if Util.parentheses?(left)
-            return range_for_parentheses(first_offense, left)
+          if Util.parentheses?(mlhs_node)
+            return range_for_parentheses(first_offense, mlhs_node)
           end
 
-          range_between(
-            first_offense.source_range.begin_pos,
-            node.loc.operator.begin_pos
-          )
+          range_between(first_offense.source_range.begin_pos,
+                        node.loc.operator.begin_pos)
+        end
+
+        def children_offenses(variables)
+          variables.flat_map { |n| unneeded_ranges(n) if n.mlhs_type? }
```

Comment:
When enumerating over `variables`, `v` seems like a natural identifier for the block. 🙂

Reply:
I was thinking of `node`. `v` is indeed better!